linux/deploy_vm/rebuild_unattend_install_iso.yml

```diff
@@ -21,245 +21,254 @@
+          - 'md5sum.txt'
+    
+    - name: "Add autoinstall to UEFI boot kernel command for Ubuntu live server"
+      ansible.builtin.replace:
+        path: "{{ unattend_iso_cache }}/grub.cfg"
+        regexp: '(.*vmlinuz)(.*)'
+        replace: "\\1 autoinstall \\2"
 
-        - name: "Update Debian grub.cfg for autoinstall"
-          ansible.builtin.blockinfile:
-            path: "{{ unattend_iso_cache }}/grub.cfg"
-            block: |
-              set default="autoinstall"
-              set timeout=5
-              menuentry "Automated installation" --id autoinstall {
-                  set background_color=black
-                  linux    /{{ debian_install_type }}/vmlinuz auto=true file=/cdrom/{{ unattend_install_file_name }} vga=788 --- quiet
-                  initrd   /{{ debian_install_type }}/gtk/initrd.gz
-              }
+    - name: "Set timeout to 5 seconds at boot menu"
+      ansible.builtin.replace:
+        path: "{{ unattend_iso_cache }}/grub.cfg"
+        regexp: 'set timeout=.*'
+        replace: "set timeout=5"
+
+    - name: "Update md5sum for UEFI boot config file"
+      ansible.builtin.shell: |
+        md5=`md5sum grub.cfg | awk '{print $1}'`
+        sed -i "/.\/boot\/grub\/grub.cfg/ s/^[^ ]*/$md5/" md5sum.txt
+      args:
+        chdir: "{{ unattend_iso_cache }}"
+    
+    - name: "set var ubuntu_bios_cfg_exist to default false" 
+      ansible.builtin.set_fact:
+        ubuntu_bios_cfg_exist: false
 
-        - name: "Update boot menu with preseed.cfg for Debian"
-          ansible.builtin.replace:
-            path: "{{ unattend_iso_cache }}/gtk.cfg"
-            regexp: '(.*)(initrd.gz )(.*)'
-            replace: "\\1\\2 auto=true file=/cdrom/{{ unattend_install_file_name }} \\3"
+    - name: Extract isolinux/txt.cfg inside ISO if exists
+      community.general.iso_extract:
+        image: "{{ src_iso_file_path }}"
+        dest: "{{ src_iso_file_dir }}"
+        files:
+          - 'isolinux/txt.cfg'
+      register: check_file_result
+      failed_when: false 
 
-        - name: "Update md5sum for Debian ISO files"
-          ansible.builtin.shell: "{{ item }}"
-          with_items:
-            - "sed -i '#./isolinux/isolinux.cfg#d' md5sum.txt"
-            - "echo \"`md5sum isolinux.cfg | awk '{print $1}'` ./isolinux/isolinux.cfg\" >>md5sum.txt"
-            - "sed -i '#./boot/grub/grub.cfg#d' md5sum.txt"
-            - "echo \"`md5sum grub.cfg | awk '{print $1}'` ./boot/grub/grub.cfg\" >>md5sum.txt"
-          args:
-            chdir: "{{ unattend_iso_cache }}"
-          register: update_initrd_output
+    - debug: var=check_file_result
 
-        - name: "Print command output for updating initrd"
-          ansible.builtin.debug: var=update_initrd_output
-      when: unattend_install_conf is match('Debian')
+    - name: "set var ubuntu_bios_cfg_exist to true or not"
+      ansible.builtin.set_fact: 
+        ubuntu_bios_cfg_exist: true
+      when: not 'msg' in check_file_result
```

Comment:
With ignore_errors, here can check check_file_result.failed or not.

Reply:
Thanks. Done